src/graphql/resolvers/box-resolver.ts

```diff
@@ -95,7 +96,11 @@ export class BoxResolver {
     @Ctx() context: GraphQLContext,
     @Info() info: GraphQLResolveInfo
   ) {
-    return context.repository.boxes.find({
+    const unconfirmedInputBoxIds = isFieldSelected(info, "beingSpent")
+      ? await context.repository.unconfirmedInputs.getUnconfirmedInputBoxIds()
```

Comment:
Great work on this! I think with optimize it by only returning selected `boxIds`.

You can add `@ConfigureLoader({ required: true })` decorator to `IBox`'s `boxId` field, so it will be selected by TypeORM even if the user don't select it on the query. And then you can use `boxes` const to get the boxIds to be used as param on `getUnconfirmedInputBoxIds()` and get more optimized results.